Client/Networking/NetworkSession.swift

```
@@ -23,6 +30,40 @@ extension URLSession: NetworkSession {
       httpVersion: "HTTP/1.1",
       headerFields: [:])!
   }
+  
+  func dataRequest(with url: URL, _ completion: @escaping (Result<NetworkSessionDataResponse, Error>) -> Void) {
+    self.dataTask(with: url) { data, response, error in
+      if let error = error {
+        completion(.failure(error))
+      } else {
+        completion(.success((data ?? Data(), response ?? self.createResponse(url: url))))
+      }
+    }.resume()
+  }
+  
+  func dataRequest(with urlRequest: URLRequest, _ completion: @escaping (Result<NetworkSessionDataResponse, Error>) -> Void) {
+    self.dataTask(with: urlRequest) { data, response, error in
+      if let error = error {
+        completion(.failure(error))
+      } else {
+        completion(.success((data ?? Data(), response ?? self.createResponse(url: urlRequest.url))))
```

should we return a failure if `data` is nil?